src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -142,7 +143,11 @@ async def setup_coursier(
         json_output_file="$1"
         shift
 
-        "$coursier_exe" fetch {repos_args} --json-output-file="$json_output_file" "$@"
+        WORKING_DIRECTORY=$(pwd)
+        ARGS=$@
```

Comment:
Maybe use bash array syntax here? Or else prefer `$*` over `$@` when trying to put all of the arguments into a single variable given how `$*` expands to a single Bash "word" when in double quotes.

See `man bash`:

```
    Special Parameters
       The  shell  treats  several parameters specially.  These parameters may only be referenced;
       assignment to them is not allowed.
       *      Expands to the positional parameters, starting from one.  When the expansion is  not
              within double quotes, each positional parameter expands to a separate word.  In con-
              texts where it is performed, those words are subject to further word  splitting  and
              pathname expansion.  When the expansion occurs within double quotes, it expands to a
              single word with the value of each parameter separated by the first character of the
              IFS  special  variable.   That is, "$*" is equivalent to "$1c$2c...", where c is the
              first character of the value of the IFS variable.  If IFS is unset,  the  parameters
              are  separated  by spaces.  If IFS is null, the parameters are joined without inter-
              vening separators.
       @      Expands to the positional parameters, starting from one.   In  contexts  where  word
              splitting  is  performed, this expands each positional parameter to a separate word;
              if not within double quotes, these words are subject to word splitting.  In contexts
              where word splitting is not performed, this expands to a single word with each posi-
              tional parameter separated by a space.  When  the  expansion  occurs  within  double
              quotes,  each  parameter expands to a separate word.  That is, "$@" is equivalent to
              "$1" "$2" ...  If the double-quoted expansion occurs within a word, the expansion of
              the  first parameter is joined with the beginning part of the original word, and the
              expansion of the last parameter is joined with the last part of the  original  word.
              When  there  are no positional parameters, "$@" and $@ expand to nothing (i.e., they
              are removed).
```

Would bash pattern replace syntax simplify this code? I.e., `${parameter/pattern/string}`

Reply:
Thanks, `$*` removes the need for the weird `echo $ARGS`. 

I tried using bash pattern replacement, it's complicated in that `$WORKING_DIRECTORY` will contain forward slashes, and it just seems generally complicated. `sed` appears to solve the problem with _much_ more readable code.

Happy to be proven wrong here though.